File: Python/Lab1.1/library.py
def text_input():
    text = ''
    print("Enter the text:\n Write @ at the beginning of a line to stop input.")
    while True:
        line = input()
        if line.find('@') == 0:
            break
        text += line + '\n'
    if text[-1] == '\n':
        text = text[:-1]
    return text


def write_file(filename, text):
    mode = input("If you want to append your input, enter a. Otherwise enter w: ")
    while (mode != 'a' and mode != 'w'):
        mode = input("Wrong letter, enter a or w: ")
    if mode == 'a':
        file = open(filename, "a")
        text = f"\n{text}"
    if mode == 'w':
        file = open(filename, "w")
    file.write(text)
    file.close()


def read_file(filename):
    with open(filename, "r") as file:
        print(f"Open {filename}\n{file.read()}\n")


def del_words(text):
    lines_list = text.split("\n")
    new_text = ""
    counter = 0
    for line in lines_list:
        words_list = line.split()
        new_line = ""
        for word in words_list:
            if not (len(word) == 1 and word.isalpha()):
                new_line += word + " "
            else:
                counter += 1
        new_text += new_line + "\n"
    return new_text, counter


def del_duplicates(text, separator):
    counter = 0
    for char in text:
        while char == separator and char * 2 in text:
            text = text.replace(char * 2, char, 1)
            counter += 1
    return text, counter


def change_file(oldFile, newFile):
    with open(oldFile, "r") as file:
        text = file.read()
        new_text, counter_space = del_duplicates(text, " ")
        new_text, counter_comma = del_duplicates(new_text, ",")
        new_text, counter_dot = del_duplicates(new_text, ".")
        new_text, counter_word = del_words(new_text)
        new_text += f"shrinked spaces - {counter_space}, commas - {counter_comma}, dots - {counter_dot}, words - {counter_word}"
        with open(newFile, "w") as new_file:
            new_file.write(new_text)

File: C#/Lab1.1/Library.cs
using System;
using System.IO;
using System.Linq;

namespace lab1._1
{
    public static class Library
    {
        public static string Input()
        {
            string text = "";
            Console.WriteLine("Enter your text:\n Write @ at the beginning of a line to stop input.\n");
            while (true) {
                string line = Console.ReadLine();
                if (line.IndexOf('@') == 0) 
                    break;
                text += line + "\n";
            }

            if (text[text.Length - 1] == '\n') text = text.Substring(0, text.Length - 1);

            return text;
        }
        
        public static void FileWrite(string text, string filename)
        {
            Console.Write("\nIf you want to append your input, enter a. Otherwise enter w: ");
            char mode = Convert.ToChar(Console.ReadLine());
            while (mode != 'a' && mode != 'w')
            {
                Console.Write("Wrong mode, enter again: ");
                mode = Convert.ToChar(Console.ReadLine());
            }
            if (mode == 'a') 
            {
                StreamWriter file = new StreamWriter(filename, true);
                file.Write($"\n{text}");
                file.Close();
            }
            if (mode == 'w')
            {
                StreamWriter file = new StreamWriter(filename, false);
                file.Write(text);
                file.Close();
            }
            
        }

        public static void ReadFile(string filename)
        {
            using (StreamReader file = new StreamReader(filename))
            {
                Console.WriteLine($"\nOpen {filename}\n{file.ReadToEnd()}");
            }
        }
        
        public static string DeleteWords(string text, ref int words_counter)
        {
            string[] lines_list = text.Split('\n');
            string new_text = "";

            foreach (var line in lines_list)
            {
                string[] words_list = line.Split();
                string new_line = "";
                foreach (var word in words_list)
                {
                    if (word.Length == 1 && word.All(ch => Char.IsLetter(ch)))
                    {
                        words_counter++;
                        continue;
                    }
                    new_line += word + " ";
                }
                new_text += new_line + "\n";
            }

            return new_text;
        }

        public static string DeleteDuplicates(string text, ref int counter_comma, ref int counter_dot, ref int counter_space)
        {
            string new_text = "";
            for (int i = 0; i < text.Length - 1; i++) {
                if (text[i] == text[i+1]){
                    if (text[i] == ',')
                        counter_comma++;
                    else if (text[i] == '.')
                        counter_dot++;
                    else if (text[i] == ' ')
                        counter_space++;
                    else
                        new_text += text[i];
                }
                else new_text += text[i];
            }

            new_text += text[text.Length - 1];

            return new_text;
        }
        
        public static void ChangeFile(string old_file, string new_file)
        {
            string text;
            int commas = 0;
            int dots = 0;
            int spaces = 0;
            int words = 0;
            
            using (StreamReader oldFile = new StreamReader(old_file))
            {
                text = DeleteDuplicates(oldFile.ReadToEnd(), ref commas, ref dots, ref spaces);
                text = DeleteWords(text, ref words);
                text += $"Removed commas - {commas}, dots - {dots}, spaces - {spaces}, words - {words}";
            }

            using (StreamWriter newFile = new StreamWriter(new_file))
            {
                newFile.Write(text);
            }
        }
    }
}